common/src/main/java/org/jboss/hal/testsuite/util/SuggestionChecker.java

```diff
@@ -0,0 +1,174 @@
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Class which pulic methods return only {@link SuggestState} objects on which the verification ca be done
+ *
+ * @author Jan Kasik <jkasik@redhat.com>
+ *         Created on 8/25/16.
+ */
+public class SuggestionChecker {
+
+    private WebDriver browser;
+    private ConfigFragment configFragment;
+    private String suggestBoxInputLabel;
+    private SuggestionResource suggestionResource;
+
+    private SuggestionChecker(Builder builder) {
+        this.browser = builder.browser;
+        this.configFragment = builder.configFragment;
+        this.suggestBoxInputLabel = builder.suggestBoxInputLabel;
+        this.suggestionResource = builder.suggestionResource;
+    }
+
+    private void clearFilter() {
+        WebElement textInput = getEditor().getText(suggestBoxInputLabel);
+        textInput.sendKeys(Keys.chord(Keys.CONTROL, "a"));
+        textInput.sendKeys(Keys.DELETE);
+    }
+
+    private List<String> getSuggestedLabelsStrings() {
+        try {
+            return Console.withBrowser(browser).openedSuggestionBoxPopUp().getLabelValues();
+        } catch (TimeoutException e) {
+            return Collections.emptyList();
+        }
+    }
+
+    /**
+     * Clears input field and sets its value to filterString
+     * @param filterString value which will be set to input field
+     * @return state of suggestion box
+     */
+    public SuggestState filterSuggestions(String filterString) {
+        clearFilter();
+        getEditor().text(suggestBoxInputLabel, filterString);
+        return new SuggestState(filterString, getSuggestedLabelsStrings());
+    }
+
+    /**
+     * Appends symbol to value of input box
+     * @param symbol value which will be appended to input box
+     * @return state of suggestion box
+     */
+    public SuggestState appendSymbolToInputField(char symbol) {
+        WebElement inputField = getEditor().getText(suggestBoxInputLabel);
+        inputField.click();
+        inputField.sendKeys(String.valueOf(symbol));
+        return new SuggestState(inputField.getAttribute("value"), getSuggestedLabelsStrings());
+    }
+
+    /**
+     * Removes symbol to value of input box
+     * @return state of suggestion box
+     */
+    public SuggestState removeSymbolFromInputField() {
+        WebElement inputField = getEditor().getText(suggestBoxInputLabel);
+        inputField.click();
+        inputField.sendKeys(Keys.BACK_SPACE);
+        return new SuggestState(inputField.getAttribute("value"), getSuggestedLabelsStrings());
+    }
+
+    private Editor getEditor() {
+        Editor editor = configFragment.getEditor();
+        if (!configFragment.isInEditMode()) {
+            editor = configFragment.edit();
+        }
+        return editor;
+    }
+
+    public class SuggestState {
+
+        private String inputFieldValue = "";
```

Comment:
No need to initialize to empty String as it is always set in constructor. Also make it final.